File: 1296-divide-array-in-sets-of-k-consecutive-numbers/1296-divide-array-in-sets-of-k-consecutive-numbers.cpp
class Solution {
public:
    bool isPossibleDivide(vector<int>& nums, int k) {
        int n = nums.size();
        if (n % k != 0) {
            return false;
        }
        map<int, int> mp;
        for (int x : nums) {
            mp[x]++;
        }
        while (!mp.empty()) {
            int curr = mp.begin()->first;
            for (int i = 0; i < k; i++) {
                if (mp[curr + i] == 0) {
                    return false;
                }
                mp[curr + i]--;
                if (mp[curr + i] < 1) {
                    mp.erase(curr+i);
                }
            }
        }
        return true;
    }
};

File: 1296-divide-array-in-sets-of-k-consecutive-numbers/1296-divide-array-in-sets-of-k-consecutive-numbers.java
class Solution {
    public boolean isPossibleDivide(int[] nums, int k) {
        int n = nums.length;
        if (n % k != 0) {
            return false;
        }
        TreeMap<Integer, Integer> mp = new TreeMap<>();
        for (int x : nums) {
            mp.put(x, mp.getOrDefault(x, 0) + 1);
        }
        while (!mp.isEmpty()) {
            int curr = mp.firstKey();
            for (int i = 0; i < k; i++) {
                if (mp.getOrDefault(curr + i, 0) == 0) {
                    return false;
                }
                mp.put(curr + i, mp.get(curr + i) - 1);
                if (mp.get(curr + i) == 0) {
                    mp.remove(curr + i);
                }
            }
        }
        return true;
    }
}